src/remembered-redis.ts

```diff
@@ -150,16 +165,56 @@ export class RememberedRedis extends Remembered {
 		}
 	}
 
-	private getSemaphore(key: string) {
-		return new Semaphore(
-			this.redis,
-			`${this.redisPrefix}REMEMBERED-SEMAPHORE:${key}`,
-			1,
-			{
-				...this.semaphoreConfig,
-				onLockLost: (err) => this.settings.onLockLost?.(key, err),
-			},
-		);
+	private getSemaphore(key: string): RememberedSemaphore {
+		if (!RememberedRedis.semaphoreLib) {
+			try {
+				RedLockClass = require('redlock').default;
+				RememberedRedis.semaphoreLib = 'redlock';
+			} catch {
+				RememberedRedis.semaphoreLib = 'redis-semaphore';
+			}
+		}
+		const { redis, semaphoreConfig } = this;
+		if (RememberedRedis.semaphoreLib === 'redis-semaphore' || !RedLockClass) {
+			return new Semaphore(
+				redis,
+				`${this.redisPrefix}REMEMBERED-SEMAPHORE:${key}`,
+				1,
+				{
+					...this.semaphoreConfig,
+					onLockLost: (err) => this.settings.onLockLost?.(key, err),
+				},
+			);
+		} else {
+			let instance = RememberedRedis.redLocksInstances.get(redis) as
+				| RedLock
+				| undefined;
+			if (!instance) {
+				instance = new RedLockClass([redis], {
```

Comment:
For this first feature version, we will not allow any configuration change for the class, right?

Reply:
For the first version, the simple fact of having redlock installed, will trigger its use. I want to remove redis-semaphore completely in the future if redlock shows a better performance